src/main/scala/akka/persistence/cassandra/CassandraPluginConfig.scala

```diff
@@ -39,8 +38,33 @@ class CassandraPluginConfig(config: Config) {
   val contactPoints = getContactPoints(config.getStringList("contact-points").asScala, port)
   val fetchSize = config.getInt("max-result-size")
 
+  private[this] val connectionPoolConfig = config.getConfig("connection-pool")
+
+  val poolingOptions = new PoolingOptions()
+    .setNewConnectionThreshold(
+      HostDistance.LOCAL,
+      connectionPoolConfig.getInt("new-connection-threshold-local"))
+    .setNewConnectionThreshold(
+      HostDistance.REMOTE,
+      connectionPoolConfig.getInt("new-connection-threshold-remote"))
+    .setMaxRequestsPerConnection(
+      HostDistance.LOCAL,
+      connectionPoolConfig.getInt("max-requests-per-connection-local"))
+    .setMaxRequestsPerConnection(
+      HostDistance.REMOTE,
+      connectionPoolConfig.getInt("max-requests-per-connection-remote"))
+    .setConnectionsPerHost(
+      HostDistance.LOCAL,
+      connectionPoolConfig.getInt("connections-per-host-core-local"),
+      connectionPoolConfig.getInt("connections-per-host-max-local"))
+    .setConnectionsPerHost(
+      HostDistance.REMOTE,
+      connectionPoolConfig.getInt("connections-per-host-core-remote"),
+      connectionPoolConfig.getInt("connections-per-host-max-remote"))
```

Comment:
we should also set poolTimeoutMillis to 0 according to https://datastax.github.io/java-driver/features/async/#known-limitations